client/src/cordova/apple/OutlineAppleLib/Sources/OutlineCatalystApp/CatalystApp.swift

```diff
@@ -53,6 +57,48 @@
             { _ in
                 appKitController._AppKitBridge_setConnectionStatus(.disconnected)
             }
+            
+            // Handle connection toggle from tray menu
+            NotificationCenter.default.addObserver(forName: NSNotification.Name("toggleConnection"),
+                                                   object: nil,
+                                                   queue: nil)
+            { _ in
+                Task {
+                    // First, ensure the window is open
+                    NotificationCenter.default.post(name: NSNotification.Name("openApplication"), object: nil)
+                    
+                    // Wait a bit for the window to open
+                    try? await Task.sleep(nanoseconds: 500_000_000) // 0.5 seconds
+                    
+                    // Try to get the web view from our stored reference first
+                    var webView = webViewReference
+                    
+                    // If we don't have a stored reference, try to find it once
+                    if webView == nil {
+                        webView = getWebView()
+                        if webView != nil {
+                            webViewReference = webView
+                        }
+                    }
+                    
+                    if let webView = webView {
+                        // Let JS handle the connection
+                        let js = """
+                        window.dispatchEvent(new CustomEvent('connectFromMenu'));
```

Comment:
This is adding extra complexity we will need to maintain. We already track the connection state and propagate events to the Typescript code with [`VpnApi.onStatusChange`](https://github.com/Jigsaw-Code/outline-apps/blob/651c15473595502eeff0342d207b01f85e73866b/client/src/www/app/outline_server_repository/vpn.ts#L48). Why can't you reuse that?

Also, why does it matter to the UI whether you are connecting from the menu?

Reply:
I believe you can simply call the VPN api directly, like we do in the Cordova plugin, and all else should work as intended, because that works if we start from the system VPN settings. Can you try that instead?

That means you should revert all changes to the Typescript code.